app/docs/src/reference/search.md

```diff
@@ -0,0 +1,184 @@
+
+- **Basic Syntax:** `InstanceType:m8g.medium` will search for m8g.medium
+  instances.
+- **Alternatives:** `InstanceType:m8g.medium|m8g.small` will search for
+  m8g.medium or m8g.large instances.
+- **Wildcards:** `InstanceType:m8g.*` will search for all m8g instances
+  regardless of size.
+
+  Wildcards can be placed anywhere in the value: `InstanceType:m*.large` will
+  match `m8g.large`, `m7g.large` and even `m7i-flex.large`.
+
+  > Tip: While building your infrastructure, you may want to find things where
+  > you did _not_ specify an attribute. For example, `!AvailabilityZone:*` will
+  > bring back instances where you did not specify an AvailabilityZone, so you
+  > can add one!
+- **Exact Matches:** `Runtime:"python3.11"` will match only the `python3.11`
+  runtime on a lambda function, but not `python3`.
+
+  You can use quotes (`"`) to pin down your search and match an exact value. If
+  you don't use quotes, things that _start with_ the value you specify are
+  matched.
+
+  Quotes will also allow you to use spaces in your search:
+  `Description:"Production Access"`.
+- **Attribute Paths:** `LaunchTemplate/Version:1` will match instances with
+  `LaunchTemplate version 1`.
+
+  Sometimes an attribute has a generic name, and you need to specify more of its
+  path. `LaunchTemplate/Version:1` is useful because it will _not_ bring in
+  every other AWS resource with a random `Version` field set to 1.
+- **Schema:** `schema:AWS::EC2::Instance`, or `schema:Instance`, will find all
+  EC2 instances.
+
+All of these features can be mixed and matched:
+`InstanceType:m8g.*|"mac1.metal"` will find `m8g` instances as well as
+`mac1.metal` instances.
+
+### Boolean Logic
+
+Sometimes you need more precise logic than just "find things matching A, B and
+C." For this, we support full boolean logic, with nesting.
+
+- **Negation:** `!InstanceType:m8g.large` will match all instances that are
+  _not_ m8g.large.
+- **Alternatives:** `Instance | Image` will match all instances and images.
+- **Grouping:** `(prod Instance) | (dev Image)` will match Instances in prod,
+  and images with "dev" in the name.
+- "And" (narrowing a search) is done by putting spaces between things. `&` is
+  supported but redundant: `prod Instance` and `prod & Instance` do the same
+  thing.
+
+## Putting It All together
+
+This search will bring back `m8g.medium` instances, and load balancers with
+`MaxSize>1`, in prod:
+
+```
+prod (schema:AWS::EC2::Instance InstanceType:m8g.* | schema:AWS::ElasticLoadBalancing::LoadBalancer !MaxSize:0|1)
+```
+
+<DocTabs tabs="CLI,AI Agent,Web Application,Public API">
+<TabPanel value="CLI">
+
+```shellscript
+$ si component search "schema:AWS::EC2::Instance" -o json
+{
+  "count": 1,
+  "components": [
+    {
+      "id": "01KBNCGCKKKHSAE23R1E2E8TMA",
+      "name": "MyNodeEC2Instance",
+      "schema": {
+        "name": "AWS::EC2::Instance"
+      }
+    }
+  ]
+}
+```
+
+</TabPanel>
+<TabPanel value="AI Agent">
```

Comment:
My only thought is that we should drop this section for now until we get it under control! It definitely shouldn't be using template-generate - but that's not a you thing to worry about :)